ToolBox/Benzinga/BenzingaDataConverter.cs

```diff
@@ -0,0 +1,321 @@
+        private IEnumerable<BenzingaNews> Process(DirectoryInfo dateDirectory)
+        {
+            foreach (var publication in dateDirectory.GetFiles("*.xml", SearchOption.TopDirectoryOnly))
+            {
+                var document = new XmlDocument();
+                document.LoadXml(File.ReadAllText(publication.FullName));
+
+                var item = JsonConvert.DeserializeObject<JObject>(JsonConvert.SerializeXmlNode(document))["rss"]["channel"]["item"];
+                // Benzinga complies with RSS standards and uses RFC 822 for DateTime
+                var publicationDate = Parse.DateTime(item["pubDate"].ToString());
+
+                var tickers = new List<JToken>();
+
+                // Only process articles that contain tickers to disk
+                if (item["bz:ticker"] == null)
+                {
+                    continue;
+                }
+
+                // Check for only a single ticker before we try to iterate
+                if (item["bz:ticker"].Type == JTokenType.Object)
+                {
+                    tickers.Add(item["bz:ticker"]);
+                }
+                else
+                {
+                    foreach (var tickerData in JArray.Parse(item["bz:ticker"].ToString()))
+                    {
+                        tickers.Add(tickerData);
+                    }
+                }
+
+                var instance = JsonConvert.DeserializeObject<BenzingaNews>(item.ToString());
+
+                // Strip all HTML tags from the article, then convert HTML entities to their string representation
+                // e.g. "<html><p>Apple&#39;s Earnings</p></html>" would become "Apple's Earnings"
+                instance.Contents = WebUtility.HtmlDecode(Regex.Replace(instance.Contents, @"<[^>]*>", " "));
+                instance.Symbols = new List<BenzingaSymbolData>();
+                instance.Metadata = new BenzingaMetadata()
+                {
+                    IsPro = item["bz:type"]["@pro"].ToString() == "1",
+                    FirstRun = item["bz:type"]["@pro"].ToString() == "1",
+                    Kind = item["bz:type"]["#text"].ToString()
+                };
+
+                foreach (var ticker in tickers)
+                {
+                    var sentiment = ticker["@sentiment"] == null ? (decimal?)null : Parse.Decimal(ticker["@sentiment"].ToString());
+                    var exchange = ticker["@exchange"] == null ? string.Empty : ticker["@exchange"].ToString();
+
+                    if (ticker["#text"] == null)
+                    {
+                        continue;
+                    }
+
+                    instance.Symbols.Add(new BenzingaSymbolData()
+                    {
+                        Exchange = exchange,
+                        Sentiment = sentiment,
+                        Symbol = new Symbol(SecurityIdentifier.GenerateEquity(ticker["#text"].ToString().Trim(), Market.USA, mapSymbol: false), ticker["#text"].ToString().Trim())
```

Comment:
I considered mapping for the tickers (and my first implementation did map them), but I discovered that the tickers in the articles are essentially already "mapped" because the news article reports on the tickers as of that date. I'm willing to be convinced that my approach is wrong here, but would like to discuss a bit if possible 👍

Reply:
Even if already mapped, was there a reason not to map them here too?
I was thinking this is something that we want because it sets the `SID` date value correctly and not to `DefaultDate`, and we want this because it ends up being part of the SID properties and used for equality. What do you think?

Reply:
I think you're right. What if a user wants to place a trade on the basket of stocks mentioned in an article? With the SID incorrectly set, it might mess up the order in some unknown way. The risks outweigh the benefits of not mapping here, so I'll update it 👍